REGRESSION (263724@main): fast/dynamic/anchor-lock.html is a flaky failure on iOS

rniwa opened the pull request:
#### 3cf82e28e0c81231a5ba1ff992ef8fd60b86ea10
<pre>
REGRESSION (263724@main): fast/dynamic/anchor-lock.html is a flaky failure on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=269492">https://bugs.webkit.org/show_bug.cgi?id=269492</a>

Reviewed by Wenson Hsieh and Simon Fraser.

Speculative flakiness fix for the test.

* LayoutTests/fast/dynamic/anchor-lock.html:

Canonical link: <a href="https://commits.webkit.org/276765@main">https://commits.webkit.org/276765@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8e1ef137408d1a4ae806c26d5d3178e45c835a7f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/45625 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/24751 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48203 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48293 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/41660 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29007 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22147 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/37369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46203 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/21824 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39349 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18510 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19225 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40438 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/3669 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/41929 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/40768 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50035 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/20613 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/17133 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44458 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/21919 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/39501 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43294 "Passed tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/22279 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6350 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/21606 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/fast/dynamic/anchor-lock.html:
```diff
@@ -19,7 +19,10 @@ <h1>Anchor Lock</h1>
     testRunner.waitUntilDone();
-    window.addEventListener('scroll', () => {
+    Promise.all([
+        new Promise((resolve) => addEventListener('scroll', resolve)),
+        new Promise((resolve) => addEventListener('load', resolve)),
```
Should we use `{ once: true }` as 3rd addEventListener option?

whsieh replied:
I suppose it doesn't hurt, but invoking promise callbacks multiple times will result in no-ops (after the first call), so the fix should be fine as it currently is.

rniwa replied:
I'd land this PR as for now since this is a speculative fix to begin with.

rniwa committed:
REGRESSION (263724@main): fast/dynamic/anchor-lock.html is a flaky failure on iOS
https://bugs.webkit.org/show_bug.cgi?id=269492

Reviewed by Wenson Hsieh and Simon Fraser.

Speculative flakiness fix for the test.

* LayoutTests/fast/dynamic/anchor-lock.html:

Canonical link: https://commits.webkit.org/276765@main
